pkg/graphql/execution_engine_v2_test.go

```diff
@@ -887,6 +887,68 @@ func TestExecutionEngineV2_Execute(t *testing.T) {
 		expectedResponse: `{"data":{"heroes":["Human","Droid"]}}`,
 	}))
 
+	t.Run("execute operation with null and omitted input variables", runWithoutError(ExecutionEngineV2TestCase{
+		schema: func(t *testing.T) *Schema {
+			t.Helper()
+			schema := `
+             type Query {
+			   heroes(names: [String!], height: String): [String!]
+		     }`
+			parseSchema, err := NewSchemaFromString(schema)
+			require.NoError(t, err)
+			return parseSchema
+		}(t),
+		operation: func(t *testing.T) Request {
+			return Request{
+				OperationName: "MyHeroes",
+				Variables:     []byte(`{"height": null}`),
+				Query: `query MyHeroes($heroNames: [String!]!, $height: String){
```

Comment:
probably will be better to have heroNames variable with nullable list as you have defined in schema `$heroNames: [String!]`

For the operation validation, it will be correct, but real upstream will return an error if required list will be null or undefined